core/output/noop.go

```diff
@@ -5,21 +5,24 @@ package output
 
 import "io"
 
-type noopOutput struct {
-	Output
-}
+// DiscardingOutput discards all output, which can be useful for testing, among other purposes.
+type DiscardingOutput struct{}
 
-func (o *noopOutput) Info(msg string)                                      {}
-func (o *noopOutput) Infof(format string, args ...interface{})             {}
-func (o *noopOutput) InfoWriter() io.Writer                                { return io.Discard }
-func (o *noopOutput) Warn(msg string)                                      {}
-func (o *noopOutput) Warnf(format string, args ...interface{})             {}
-func (o *noopOutput) WarnWriter() io.Writer                                { return io.Discard }
-func (o *noopOutput) Error(err error, msg string)                          {}
-func (o *noopOutput) Errorf(err error, format string, args ...interface{}) {}
-func (o *noopOutput) ErrorWriter() io.Writer                               { return io.Discard }
-func (o *noopOutput) StartOperation(status string)                         {}
-func (o *noopOutput) EndOperation(success bool)                            {}
-func (o *noopOutput) Result(result string)                                 {}
-func (o *noopOutput) ResultWriter() io.Writer                              { return io.Discard }
-func (o *noopOutput) WithValues(keysAndValues ...interface{}) Output       { return o }
+// Convention used to verify, at compile time, that DiscardingOutput implements the Output interface.
+var _ Output = DiscardingOutput{}
+
+func (o DiscardingOutput) Info(msg string)                                      {}
+func (o DiscardingOutput) Infof(format string, args ...interface{})             {}
+func (o DiscardingOutput) InfoWriter() io.Writer                                { return io.Discard }
+func (o DiscardingOutput) Warn(msg string)                                      {}
+func (o DiscardingOutput) Warnf(format string, args ...interface{})             {}
+func (o DiscardingOutput) WarnWriter() io.Writer                                { return io.Discard }
+func (o DiscardingOutput) Error(err error, msg string)                          {}
+func (o DiscardingOutput) Errorf(err error, format string, args ...interface{}) {}
+func (o DiscardingOutput) ErrorWriter() io.Writer                               { return io.Discard }
+func (o DiscardingOutput) StartOperation(status string)                         {}
+func (o DiscardingOutput) EndOperation(success bool)                            {}
+func (o DiscardingOutput) Result(result string)                                 {}
+func (o DiscardingOutput) ResultWriter() io.Writer                              { return io.Discard }
+func (o DiscardingOutput) WithValues(keysAndValues ...interface{}) Output       { return o }
+func (o DiscardingOutput) V(level int) Output                                   { return o }
```

Comment:
This will change the behaviour of code like this:
```go
o := output.NewInteractiveShell(os.Stdout, os.Stderr, 1)
o2 := o.V(2)
o2.V(1).Info("info with verbosity=1)
```
Before this PR the info was printed, after this PR it will no longer be printed.
Not sure if we actually want or need to be able to reduce verbosity using `V()`? I think it's fine, just wanted to point this out.

Reply:
Thanks for the review @floridoo.

I didn't intend to change any behavior. I'll give this some thought. I welcome your suggestions (or commits).

Reply:
I added a commit that adds `DiscardingOutput` while keeping `noopOutput`. What do you think?

Reply:
> Not sure if we actually want or need to be able to reduce verbosity using V()?

I think we do _not_ want to reduce verbosity using `V()`. Verbosity is a value set by the user, and it means "I allow output up to this level". The message level is set by the code using `V()`.

My original change made the level change the verbosity, which was incorrect. Your additional commit fixes that. Thanks!

I added a unit test to verify this.